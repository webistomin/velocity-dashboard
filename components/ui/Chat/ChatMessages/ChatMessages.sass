+b('chat-messages')
  border-bottom: rem(1px) solid rgba(var(--color-primary), 0.2)

  +media($screen-xl)
    border-bottom: none
    border-left: rem(1px) solid rgba(var(--color-primary), 0.2)
    border-right: rem(1px) solid rgba(var(--color-primary), 0.2)

  +e('content')
    box-sizing: border-box
    padding: rem(24px)
    height: calc(100vh - 22.6rem)
    overflow: auto
    +scrollbar(0.5rem, rgb(var(--color-primary)), rgb(var(--color-secondary)))
    +scroll-overflow()

    +media($screen-lg)
      height: calc(100vh - 20rem)

  +e('controls')
    box-sizing: border-box
    padding: rem(24px)
    height: 12rem
    background-color: rgb(var(--color-white))

  +e('input')
    box-sizing: border-box
    padding: rem(20px) rem(70px)
    width: 100%
    height: 7rem
    border-radius: 5px
    background-color: rgba(var(--color-lavender), 0.2)
    border: rem(1px) solid rgb(var(--color-lavender))
    font: var(--text-input)
    color: rgb(var(--color-primary))

  +e('inputs')
    position: relative

  +e('emoji-picker')
    position: absolute
    width: 6rem
    height: 100%
    top: 0
    left: 0
    color: rgba(var(--color-primary), 0.5)
    transition: color 0.2s ease-out

    &:hover
      color: rgb(var(--color-primary))

  +e('attach')
    position: absolute
    width: 6rem
    height: 100%
    top: 0
    right: 0

  +e('attach-label')
    display: flex
    align-items: center
    justify-content: center
    width: inherit
    height: inherit
    color: rgba(var(--color-primary), 0.5)
    transition: color 0.2s ease-out

    &:hover
      color: rgb(var(--color-primary))

  +e('item')
    display: flex
    position: relative

    &:not(:last-of-type)
      margin-bottom: rem(24px)

    +m('other')
      .chat-messages__holder
        margin-left: rem(8px)

      .chat-messages__group-item
        background-color: rgb(var(--color-white))

    +m('own')
      flex-direction: row-reverse
      text-align: right

      .chat-messages__message
        text-align: left

      .chat-messages__holder
        margin-right: rem(8px)

      .chat-messages__group-item
        background-color: rgb(var(--color-primary))
        color: rgb(var(--color-white))

  +e('name')
    font-weight: 400
    color: rgb(var(--color-gray-3))
    margin-bottom: rem(4px)

  +e('group-item')
    padding: rem(18px) rem(18px) rem(6px)

    &:first-of-type
      border-top-left-radius: 17px
      border-top-right-radius: 17px

    &:not(:last-of-type)
      margin-bottom: rem(6px)

    &:last-of-type
      border-bottom-left-radius: 17px
      border-bottom-right-radius: 17px

  +e('date')
    display: inline-block
    padding-top: rem(12px)
    font-weight: 500

  +e('avatar')
    position: sticky
    top: 0
    flex-shrink: 0

  +e('list')
    outline: none

body.using-keyboard
  .chat-messages__list:focus
    outline: rem(1px) solid rgb(var(--color-primary))

  .chat-messages__attach-input:focus ~ .chat-messages__attach-label
    outline: rem(1px) solid rgb(var(--color-primary))

