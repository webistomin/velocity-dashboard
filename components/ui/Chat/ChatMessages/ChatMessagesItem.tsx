import Vue, { CreateElement, RenderContext, VNode, PropType } from 'vue';

import { BaseThumbnail } from 'components/base/BaseThumbnail/BaseThumbnail';
import { IChatMessage } from 'common/types/chat/chat-list';

export interface IChatMessagesItemProps {
  index: number;
  source: object;
}

export const ChatMessagesItem = Vue.extend({
  functional: true,
  props: {
    index: {
      type: Number,
      default: 0,
    },
    source: {
      type: Object,
      default: {} as PropType<IChatMessage>,
    },
  },
  render(_h: CreateElement, ctx: RenderContext<IChatMessagesItemProps>): VNode {
    const { staticClass, class: cls } = ctx.data;
    const { sender, messages } = ctx.props.source as IChatMessage;
    return (
      <div class={`chat-messages__item chat-messages__item_${sender.type} list-item ${staticClass || ''} ${cls || ''}`}>
        <BaseThumbnail class='chat-messages__avatar' image={sender.avatar} size='s' alt={sender.name} />
        <div class='chat-messages__holder'>
          <strong class='chat-messages__name'>{sender.name}</strong>
          <ul class='chat-messages__group-messages list'>
            {messages.map((message) => {
              return (
                <li class='chat-messages__group-item list-item'>
                  <p class='chat-messages__message paragraph'>{message.message}</p>
                  <time class='chat-messages__date'>{message.time}</time>
                </li>
              );
            })}
          </ul>
        </div>
      </div>
    );
  },
});
