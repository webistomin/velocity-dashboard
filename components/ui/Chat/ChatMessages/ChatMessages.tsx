import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import './ChatMessages.sass';
import { BaseThumbnail } from 'components/base/BaseThumbnail/BaseThumbnail';

@Component({
  name: 'ChatMessages',
})
export default class ChatMessages extends VueComponent {
  public render(): VNode {
    return (
      <div class='chat-messages'>
        <div class='chat-messages__content'>
          <ul class='chat-messages__list list'>
            <li class='chat-messages__item chat-messages__item_own list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Anna</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_other list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Nora</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_own list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Anna</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_other list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Nora</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_own list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Anna</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_other list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Nora</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_own list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Anna</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
            <li class='chat-messages__item chat-messages__item_other list-item'>
              <BaseThumbnail class='chat-messages__avatar' image='/img/avatar.png' size='s' alt='Me' />
              <div class='chat-messages__holder'>
                <strong class='chat-messages__name'>Nora</strong>
                <ul class='chat-messages__group-messages list'>
                  <li class='chat-messages__group-item list-item'>
                    <p class='chat-messages__message paragraph'>
                      Hello John, thank you for calling Provide Support. How may I help you?
                    </p>
                    <time class='chat-messages__date'>18:54</time>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
        <div class='chat-messages__controls'>
          <div class='chat-messages__inputs'>
            <button class='chat-messages__emoji-picker btn' type='button'>
              <svg-icon name='icon-smile' width={20} height={20} />
            </button>
            <input type='text' class='chat-messages__input' placeholder='Message' />
            <div class='chat-messages__attach'>
              <label for='chat-attach' class='chat-messages__attach-label label'>
                <svg-icon name='icon-attach' width={23} height={20} />
              </label>
              <input type='file' id='chat-attach' class='chat-messages__attach-input visually-hidden' />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
