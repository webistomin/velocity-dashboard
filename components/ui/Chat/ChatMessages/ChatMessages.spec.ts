import { shallowMount } from '@vue/test-utils';
import ChatMessages from './ChatMessages';

describe('ChatMessages', () => {
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(ChatMessages, {
      propsData: {
        messages: [
          {
            id: 'message-1',
            sender: {
              type: 'other',
              avatar: '/img/avatar.png',
              name: 'Alexey Istomin',
            },
            messages: [
              {
                message: 'Hi, I have trouble with trip #12422',
                time: '02-06-2020',
              },
            ],
          },
        ],
        ...propsData,
      },
      stubs: ['VirtualList', 'svg-icon'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
