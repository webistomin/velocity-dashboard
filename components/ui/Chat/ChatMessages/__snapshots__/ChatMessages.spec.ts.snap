// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessages Matches snapshot 1`] = `
"<div class=\\"chat-messages\\">
  <div class=\\"chat-messages__content\\">
    <vuecomponent-stub datakey=\\"id\\" datasources=\\"[object Object]\\" datacomponent=\\"function VueComponent (options) {
      this._init(options);
    }\\" keeps=\\"30\\" estimatesize=\\"50\\" direction=\\"vertical\\" start=\\"0\\" offset=\\"0\\" topthreshold=\\"0\\" bottomthreshold=\\"0\\" roottag=\\"div\\" wraptag=\\"div\\" wrapclass=\\"chat-messages__scroller-wrapper\\" itemtag=\\"div\\" itemclass=\\"chat-messages__scroller-item\\" headertag=\\"div\\" headerclass=\\"\\" footertag=\\"div\\" footerclass=\\"\\" tabindex=\\"0\\" class=\\"chat-messages__list list\\"></vuecomponent-stub>
  </div>
  <div class=\\"chat-messages__controls\\">
    <div class=\\"chat-messages__inputs\\"><button type=\\"button\\" class=\\"chat-messages__emoji-picker btn\\">
        <svg-icon-stub name=\\"icon-smile\\" width=\\"20\\" height=\\"20\\"></svg-icon-stub>
      </button><input type=\\"text\\" placeholder=\\"Message\\" class=\\"chat-messages__input\\">
      <div class=\\"chat-messages__attach\\"><input type=\\"file\\" id=\\"chat-attach\\" class=\\"chat-messages__attach-input visually-hidden\\"><label for=\\"chat-attach\\" class=\\"chat-messages__attach-label label\\">
          <svg-icon-stub name=\\"icon-attach\\" width=\\"23\\" height=\\"20\\"></svg-icon-stub>
        </label></div>
    </div>
  </div>
</div>"
`;
